confluent_kafka_helpers/consumer.py

```diff
@@ -107,3 +109,61 @@ def _get_topics(self, config):
     @property
     def is_auto_commit(self):
         return self.config.get('enable.auto.commit', True)
+
+
+class AvroRawConsumer(Consumer):
+    """
+    By default the Confluent AvroConsumer decode all messages in the partition.
+
+    This consumer uses a lazy approach, it doesn't decode the messages, just
+    provide the methods so we can do it manually.
+
+    We use this approach, because we want to check the key messages before
+    decoding the message, this will avoid performance issues.
+    """
+    def __init__(self, config, schema_registry=None):
+        schema_registry_url = config.pop("schema.registry.url", None)
+        if schema_registry is None:
+            if schema_registry_url is None:
+                raise ValueError("Missing parameter: schema.registry.url")
+            schema_registry = CachedSchemaRegistryClient(
+                url=schema_registry_url
+            )
+        elif schema_registry_url is not None:
+            raise ValueError("Cannot pass schema_registry along with schema.registry.url config")  # noqa
+
+        self._lazy_loading = config.pop('lazy_loading', False)
+        super(AvroRawConsumer, self).__init__(config)
+        self._serializer = MessageSerializer(schema_registry)
+
+    def poll(self, timeout=None):
+        """
```

Comment:
I don't think this docstring adds any value. Also we don't use restructured format